### PR TITLE
Feature/create user entity

### PR DESCRIPTION
**Pull request by fabianrupin195**

### CODE REVIEW

File: src/Controller/UserController.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        $user = new User();
+
+        $user->setFirstname('Bernard')
+            ->setLastname('Toto')
+            ->setPseudo('BerToto'.rand(1, 1000))
+            ->setEmail('berToto'.rand(1, 1000))
+            ->setRole('users')
+            ->setPassword('motdepasse')
+            ->setCreatedAt(new \DateTime('now'));
```
Comment (juuuuuu) [inline src/Controller/UserController.php]: You should put the `;` on a newline, indented on the `$user`.  
This way, if you add a new `setXxx()` method, the diff will only show the added line only.

File: src/DataFixtures/ORM/fixtures.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+App\Entity\User:
```
Comment (juuuuuu) [inline src/DataFixtures/ORM/fixtures.yml]: You defined in your `.editoconfig` YAML indent size at 4 spaces. In this file you only have 2 spaces.

File: src/Entity/User.php [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    private $role;
+
+    /**
+     * @ORM\Column(type="string", length=255, unique=true)
+     */
+    private $pseudo;
```
Comment (juuuuuu) [inline src/Entity/User.php]: Maybe you could add Constraint Validation on the properties of this Entity.  
See https://symfony.com/doc/current/validation.html

File: src/Entity/User.php [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+     */
+    private $created_at;
+
+    /**
+     * @ORM\Column(type="datetime", nullable=true)
+     */
+    private $updated_at;
+
+    /**
+     * @ORM\Column(type="string", length=255, unique=true)
```
Comment (juuuuuu) [inline src/Entity/User.php]: `type="string"` and `length=255` are the default values, you can ommit it.

File: src/Entity/User.php [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+        return $this;
+    }
+
+    public function getCreatedAt(): ?\DateTimeInterface
+    {
+        return $this->created_at;
+    }
+
+    public function setCreatedAt(\DateTimeInterface $created_at): self
```
Comment (juuuuuu) [inline src/Entity/User.php]: You could add the Timestampable feature for `updatedAt` and `createdAt`.  
See https://github.com/Atlantic18/DoctrineExtensions/blob/master/doc/timestampable.md

File: .editorconfig [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+[Makefile]
+indent_style = tab
+indent_size = 8
+
+[*.php, *.yml]
+indent_size = 4
+
+[*.md]
+trim_trailing_whitespace = false
```
[the remark this answers is not in the document]
Comment (fabianrupin195) [reply .editorconfig]: :+1:

File: .gitignore [hunk 7]
```diff
@@ -24,3 +24,8 @@ frontend-assets/html
 public/css
 public/fonts
 public/js
+
+###> friendsofphp/php-cs-fixer ###
+/.php_cs
```
[the remark this answers is not in the document]
Comment (fabianrupin195) [reply .gitignore]: Symfony ignore it automatically :)

File: src/Controller/UserController.php [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class UserController extends AbstractController
+{
+    /**
+     * @Route("/user/sign-up", name="user-sign-up")
+     *
+     * @param EntityManagerInterface $entityManager
```
[the remark this answers is not in the document]
Comment (fabianrupin195) [reply src/Controller/UserController.php]: done

File: src/Controller/UserController.php [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+     *
+     * @return Response
```
[the remark this answers is not in the document]
Comment (fabianrupin195) [reply src/Controller/UserController.php]: done

File: src/Controller/UserController.php (hunk 1 above)
Comment (fabianrupin195) [reply src/Controller/UserController.php]: done

File: src/Entity/User.php (hunk 4 above)
Comment (fabianrupin195) [reply src/Entity/User.php]: done. I used the entity maker, so it did this automatically.

File: src/Entity/User.php (hunk 5 above)
Comment (fabianrupin195) [reply src/Entity/User.php]: yes but I need to install  StofDoctrineExtensionsBundle and I think it's forbidden : 'L’utilisation de bundles tiers est interdite sauf pour les données initiales'

File: src/DataFixtures/ORM/fixtures.yml (hunk 2 above)
Comment (fabianrupin195) [reply src/DataFixtures/ORM/fixtures.yml]: yes, but I don't know why...

File: src/Entity/User.php (hunk 5 above)
Comment (juuuuuu) [reply src/Entity/User.php]: You are totally right. That's absurd, but, nevermind. Sorry for the noise.

File: .gitignore (hunk 7 above)
[the remark this answers is not in the document]
Comment (juuuuuu) [reply .gitignore]: But, do you really need a `.php_cs.dist` file? Does this file will change depending on the environment? I don't think so.  
It's not a big deal, but having `.dist` and ignoring the non `.dist` file means you should have a different file on some different environment. About PHP CS Fixer, this file should not be different depending on the environment.

File: src/Entity/User.php [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+    /**
+     * @ORM\Id()
+     * @ORM\GeneratedValue()
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column()
+     * @Assert\Choice({"User", "Admin"})
```
Comment (juuuuuu) [inline src/Entity/User.php]: It's a good pratice to have roles names starting with `ROLE_*`.  
See: https://symfony.com/doc/current/security.html

File: src/Entity/User.php [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+     * @ORM\Column(unique=true)
+     * @Assert\Email()
+     */
+    private $email;
+
+    /**
+     * @ORM\Column(unique=true)
+     * @Assert\Length(
+     *     min = 3,
+     *     max = 15
```
Comment (juuuuuu) [inline src/Entity/User.php]: If you set a maximum lenght, you will have to add information about that constraint in the frontend form. I don't think that it's recommended.   
It's up to you.

File: src/Entity/User.php [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+     *     max = 25
+     * )
+     */
+    private $password;
+
+    /**
+     * @ORM\Column(type="datetime")
+     * @Assert\DateTime()
+     */
+    private $created_at;
```
Comment (juuuuuu) [inline src/Entity/User.php]: CamelCase here: `createdAt`. PHP Cs Fixer should have told you that. Same for `updatedAt`.

**Pull request merged**